helix-core/src/main/java/org/apache/helix/controller/stages/IntermediateStateCalcStage.java

```diff
@@ -239,11 +239,14 @@ private void validateMaxPartitionsPerInstance(ClusterEvent event,
           // in this instance
           partitionCount++;
           if (partitionCount > maxPartitionPerInstance) {
+            // Enable maintenance rebalancer for this pipeline
+            cache.enableMaintenanceMode();
```

Comment:
This could not be very useful. Because it is the stage after BestPossible. And the flag will be read in the cache read stage. Since we are not changing the place to add ZNode of maintenance mode. Let's try to minimize the code change.

Reply:
Yes you're right. The maintenance rebalancer is used in the best possible stage. I'll keep it as is.